### PR TITLE
libcap/2.73: Bump version

### PR DESCRIPTION
**Pull request by bobrofon**
### Summary
Changes to recipe:  **libcap/2.73**

#### Motivation
A new version of libcap is available. Full diff between 2.70 and 2.73: https://git.kernel.org/pub/scm/libs/libcap/libcap.git/diff/?id=libcap-2.73&id2=libcap-2.70

#### Details
Conan-specific patches are the same as for the previous versions. They are only rebased on version 2.73.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/libcap/all/patches/2.73/0001-libcap-Remove-hardcoded-fPIC.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+diff --git a/libcap/Makefile b/libcap/Makefile
+index 5f5c601..02be223 100644
+--- a/libcap/Makefile
++++ b/libcap/Makefile
+@@ -19,7 +19,7 @@ PSXFILES=../psx/psx ../psx/wrap/psx_wrap
+ PSXMAGICOBJ=psx_magic.o
+ 
+ # Always build libcap sources this way:
+-CFLAGS += -fPIC -D_LIBPSX_PTHREAD_LINKAGE
++CFLAGS += -D_LIBPSX_PTHREAD_LINKAGE
```
Comment (ErniGH) [inline recipes/libcap/all/patches/2.73/0001-libcap-Remove-hardcoded-fPIC.patch:20]: Hi @bobrofon, Why is this line added if nothing similar was added in the previous patch?  I couldn’t find anything 
 upstream—could you help me? 😄 

https://github.com/conan-io/conan-center-index/blob/master/recipes/libcap/all/patches/2.57/0001-libcap-Remove-hardcoded-fPIC.patch#L21

Comment (bobrofon) [reply recipes/libcap/all/patches/2.73/0001-libcap-Remove-hardcoded-fPIC.patch:20]: In the previous patch, the `-fPIC` flag was removed from `CFLAGS`. In this patch, the `-fPIC` flag is also removed from `CFLAGS`. Honestly, I don't see any difference.

```diff
-CFLAGS += -fPIC
```
vs
```diff
-CFLAGS += -fPIC -D_LIBPSX_PTHREAD_LINKAGE
+CFLAGS += -D_LIBPSX_PTHREAD_LINKAGE
```

Comment (bobrofon) [reply recipes/libcap/all/patches/2.73/0001-libcap-Remove-hardcoded-fPIC.patch:20]: The associated change in the upstream: https://git.kernel.org/pub/scm/libs/libcap/libcap.git/diff/libcap/Makefile?id=libcap-2.73&id2=libcap-2.70
```diff
-CFLAGS += -fPIC
+CFLAGS += -fPIC -D_LIBPSX_PTHREAD_LINKAGE
```

**Review by ErniGH — APPROVED** [review]
LGTM Thanks!

**Review by jcar87 — APPROVED** [review]
